test/solarwinds_apm/init_test/init_7_test.rb

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2024 SolarWinds, LLC.
+# All rights reserved.
+
+require 'initest_helper'
+
+describe 'solarwinds_apm_init_7' do
+  it 'when_there_is_problem_loading_solarwinds_apm_io' do
+    puts "\n\033[1m=== TEST RUN: #{RUBY_VERSION} #{File.basename(__FILE__)} #{Time.now.strftime('%Y-%m-%d %H:%M')} ===\033[0m\n"
+
+    log_output = StringIO.new
+    SolarWindsAPM.logger = Logger.new(log_output)
+
+    ENV['SW_APM_REPORTER'] = 'file'
+
+    require './lib/solarwinds_apm'
+    assert_includes log_output.string, 'Error occurs while loading solarwinds_apm. SolarWinds APM disabled.'
+    assert_includes log_output.string, 'Error: cannot load such file'
+    assert_includes log_output.string, 'See: https://documentation.solarwinds.com/en/success_center/observability/default.htm#cshid=config-ruby-agent'
+
+    assert_nil(defined?(SolarWindsAPM.loaded))
+
+    # we don't load noop in case that the oboe c lib is missing
```

Comment:
OK yes so the `.so` file doesn't exist (answers my question above). But we'll need to think more about the not loading noop part--the idea is even if our custom distro did not fully initialize (including loading core library, set up OTel SDK, etc) for whatever reason, whether explicitly disabled via config, or due to missing/invalid service key, wrong platform, missing/bad c-extension `.so`, we should strive to not cause an issue if the customer's application code has [custom instrumentation calls](https://github.com/solarwinds/apm-ruby/blob/NH-75734/README.md#custom-instrumentation).

Reply:
Just tested. If the core library is causing issue, then it will break custom instrumentation calls (since it doesn't require api part code)

Reply:
Same when SolarWindsAPM.loaded is false. It will break user's code if they have the custom instrumentation calls